server/src/main/java/com/linecorp/centraldogma/server/internal/mirror/DefaultMirroringServicePlugin.java

```diff
@@ -47,12 +50,30 @@ public synchronized CompletionStage<Void> start(PluginContext context) {
         DefaultMirroringService mirroringService = this.mirroringService;
         if (mirroringService == null) {
             final CentralDogmaConfig cfg = context.config();
+            final Optional<PluginConfig> optional =
+                    cfg.pluginConfigs().stream().filter(plugin -> configType().isInstance(plugin))
+                       .findFirst();
+            final int numThreads;
+            final int maxNumFilesPerMirror;
+            final long maxNumBytesPerMirror;
+
+            final MirroringServicePluginConfig mirroringServicePluginConfig =
+                    (MirroringServicePluginConfig) optional.orElse(null);
+            if (mirroringServicePluginConfig != null) {
+                numThreads = mirroringServicePluginConfig.numMirroringThreads();
+                maxNumFilesPerMirror = mirroringServicePluginConfig.maxNumFilesPerMirror();
+                maxNumBytesPerMirror = mirroringServicePluginConfig.maxNumBytesPerMirror();
+            } else {
+                numThreads = MirroringServicePluginConfig.INSTANCE.numMirroringThreads();
```

Comment:
Question) If `cfg.pluginConfigs()` doesn't have a `MirroringServicePluginConfig`, I thought this plugin wouldn't be added in the first place since it wouldn't be in the following list.

https://github.com/line/centraldogma/blob/326b5099ba973b937add2a99ae82a21cbcef4c3c/server/src/main/java/com/linecorp/centraldogma/server/PluginGroup.java#L86-L90

I'm probably missing something, but do you mind explaining when this scenario can occur?

Reply:
> If cfg.pluginConfigs() doesn't have a MirroringServicePluginConfig, I thought this plugin wouldn't be added in the first place since it wouldn't be in the following list.

Had a chat about this with @ikhoon and we've decided to enable the plugin if there's no configuration specified in the `plugins` property.

Do you think we shouldn't add the plugin if the `PluginConfig` is missing? 🤔

Reply:
I think I just missed the following line. No objection to enabling plugins without a config specified 👍 

https://github.com/line/centraldogma/blob/326b5099ba973b937add2a99ae82a21cbcef4c3c/server/src/main/java/com/linecorp/centraldogma/server/plugin/Plugin.java#L58

Reply:
Do you think it will be less confusing if we use `pluginConfigs` instead of `plugins`? 🤔 
```
{
  "dataDir": "./data",
  ...
  "pluginConfigs": [
    {
      "type": "com.linecorp.centraldogma.server.mirror.MirroringServicePluginConfig",
      "enabled": true,
      "numMirroringThreads": 16,
      "maxNumFilesPerMirror": 8192
    }  
  ]
}
```

Reply:
Good point 👍  I think so (I assume you are referring to the naming exposed publicly)

Reply:
Yeah, the name in the configuration file. 😉 
I thought the name `plugins` might have caused confusion because it's not plugins themselves, but rather plugin configurations, which can be omitted.

Thanks, fixed. 😉